### ISSUE TITLE
landmarks in ShadowRoot not recognised

### ISSUE THREAD

**Issue by cookingmaia** [xi/a11y-outline#14]
Thank you for your great extension!
An area for improvement:
On websites using ShadowRoot, landmarks inside ShadowRoot are not recognised whereas screenreaders interpret them.
Example: 
https://www.francetravail.fr/accueil/ has at least a banner, a navigation, a search, a main and a contentinfo regions, but as most of them are in ShadowRoot elements, the A11y Outline extension only detects the main region.

**Comment by xi** [comment]
Thanks for the report. From what I understand, trees below a ShadowRoot should indeed be included in the accessibility tree.

I am not sure whether it is feasible to implement this.

We already need to re-implement `querySelector()` because of `aria-owns`. Adding ShadowRoot support should be possible. The important APIs are `ShadowRoot.host` to walk up and `el.shadowRoot.childNodes` to walk down the tree.

The bigger issue are selectors for implicit roles, e.g. `aside:not(article *, aside *, nav *, section *)`. These won't work across ShadowRoots. We either have to re-impelement them with a different mechanism or live with the fact that they don't work.

**Comment by xi** [comment]
@cookingmaia do you think you could contribute a test case that can be integrated into the existing [test suite](https://github.com/xi/aria-api/tree/main/test)?

**Comment by xi** [comment]
I just published an update that adds very basic support for shadow DOM. That is: the extension can now see elements inside of a ShadowRoot, but a lot of things don't work properly yet. I am not sure if/when I will have time for more on this. So contributions welcome!
